ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/block/BlockManager.java

```diff
@@ -60,6 +62,8 @@
   private final Subscribers<ImportedBlockListener> receivedBlockSubscribers =
       Subscribers.create(true);
 
+  private BlockImportMetrics blockImportMetrics;
```

Comment:
Should mark this `final` and either explicitly assign `null` or make it an `Optional`. I'd be tempted to make it an `Optional` just to be completely clear even though we'll wind up using `.orElseGet()` in the one place it's used.  Or possibly we remove `blockImportPerformanceEnabled` field and use the presence or absence of this optional as the flag.